### ISSUE TITLE
User comments

### ISSUE THREAD

**Issue by fohristiwhirl** [fohristiwhirl/nibbler#10]
- [ ] Will there be drag and drop functionality?
- [ ] I don't find the ranking node head very useful aside from finding the best move (which colour already does), so maybe it would be better if it showed the probability (in percentage perhaps, since it's a bit more readable) and node count.
- [ ] It would also be nice to have a sort of 'threats' command, where it flips side to move to show 'threats', kinda like making a passive move, which is often very useful in analysis.
- [ ] Hover over arrow/arrow head highlights the PV, and vice-versa.
- [ ] Play out the pv when you hover over a stone placement.
- [ ] Hover over piece prioritizes their move arrows (sets them visually 'on top'), or maybe even turns off every other arrow after a small delay? This would be useful for when the move doesn't show under the current Move display setting.
- [ ] Arrow alpha.
- [x] If FEN wasn't changed, don't reset analysis or PGN.
- [ ] Show speed and total nodecount somewhere.
- [ ] Mousewheel scrolls through PGN.
- [ ] Key rebinding? ctrl-H is a tough reach.

**Comment by maxchess** [comment]
Any chance of board co-ordinates please - my eyesight/recognition not what it used to be.
Many thanks for Nibbler.

**Comment by fohristiwhirl** [comment]
Around the edges, or on the squares?

**Comment by maxchess** [comment]
Around the edges please but will be happy with whatever is easiest to code - if on the board maybe a little bolder than normal.

**Comment by maxchess** [comment]
![Screen Shot 2019-06-24 at 12 29 24 PM](https://user-images.githubusercontent.com/19890516/60035827-be569580-967b-11e9-8215-88caf2e1e256.png)

**Comment by maxchess** [comment]
You have a dark/black background so white co-ordinates will do nicely.

**Comment by koedem** [comment]
Suggestion: Option to see score from White POV. (no matter who to move) Basically every GUI has that per default. (except for Arena I think, where it still is an option)

Maybe add a Q+U in the technical info bit, for the lazy who don't want to divide the score by 50 and then add it to U. :D

**Comment by maxchess** [comment]
Thanks for v0.6.1 !!

**Comment by fohristiwhirl** [comment]
Don't thank me too much, I didn't add coordinates, got a lot of things on the todo list as you see. :)

But I have a fairly frenetic release schedule.

**Comment by maxchess** [comment]
No worries - I can wait .. :)

**Comment by AlexisOlson** [comment]
Suggestions:

1. Put N, P, U on a new line  (or before move list) so they align with each other.
2. Enable sort move list by Q, N, P, U.
3. Cross-highlight the board when hovering over move in the right pane (like hovering on the board position cross-highlights the other way).

**Comment by maxchess** [comment]
Fast & Furious (maybe not furious) !!
Thanks !

**Comment by mgemard** [comment]
I can do  "Show opening names."

**Comment by mgemard** [comment]
I will look at "Search from openings by name to immediately set up position." as well since it is kinda related.

**Comment by fohristiwhirl** [comment]
Hmm - I'm not exactly soliciting pull requests, this is mostly for my own tracking. Not that everything listed here is gonna be done.

**Comment by fohristiwhirl** [comment]
@mgemard - I hope you saw that message... no guarantees on merging things...

**Comment by mgemard** [comment]
Yes I saw the message, thank you.

I came back because I have an idea for a new feature: 

- Add arrow controls for the game (go at the start of the game, go at the end of game, next move, previous move). Look at [this picture](https://image.shutterstock.com/display_pic_with_logo/228103603/1449070667/stock-vector-media-player-controls-icon-set-1449070667.jpg) if you dont see what I mean.

**Comment by mgemard** [comment]
For the list of opening name with FEN, you can look here: https://github.com/ornicar/scalachess/tree/master/src/main/scala/opening
It is what lichess is using.

Do you know if there exist a database available we could use that list all of the opening name with FEN?

**Comment by fohristiwhirl** [comment]
It's probably not a feature I actually want - it seems like significant added work / code / complexity for not a huge amount of benefit...

**Comment by mgemard** [comment]
For the list of opening name with FEN, you can look here: https://github.com/ornicar/scalachess/tree/master/src/main/scala/opening
It is what lichess is using.

Do you know if there exist a database available we could use that list all of the opening name with FEN?

**Comment by fohristiwhirl** [comment]
No I don't know of any such database. But to repeat, I discourage you from any course of action that depends on me accepting pull requests. Of course, you're free to have your own fork, and I might even accept PRs if you're lucky, but *caveat actor*.

**Comment by al75an** [comment]
Hi, 
Thank you very much for this fantastic chess Gui,
Could you please add the option to add a second engine ( like Stockfish) for analysis.
I love the Gui and lc0 but I have a weak GPU. I think this will help lots of people like me to get better Chess analyzes

**Comment by AlexisOlson** [comment]
@al75an Stockfish has a fundamentally different search algorithm not compatible with Nibbler. There are other GUIs better suited for Stockfish.

**Comment by al75an** [comment]
Thank you Alexis, I am using Chessbase and Scid vs Pc but I am not happy with both GUI.
I love nibbler and I thought that could possible.
thank you for letting me know

**Comment by fohristiwhirl** [comment]
Thing is, traditional alpha/beta engines aren't able to give the sort of multi-variation output that Nibbler requires, without sacrificing their strength greatly.

**Comment by conqueran** [comment]
I know this isn't the place for comments but really love Nibbler. I like playing against the lower rated leelas and appreciate how easy it is to use.

**Comment by fohristiwhirl** [comment]
Thanks!

**Comment by ipostr08** [comment]
Allow higher CPuct settings and add fpu-reduction option. Currently, the net 42850 fixates on one line with something like 98% of nodes searched when another move would be seen as superior with more nodes.

**Comment by kubiknubika** [comment]
add loop to selfplay.
add training to selfplay.
add tournament or serial games with pgn-pack.
add graph-save: jpeg/pgn-annotation.
add double-weight use for selfplay.
960-chess set seed.
sygyzy for self-play (or sf will play in 5+ eval positions to minimize the useless endgame moves by lc0, or autoresign).
preset openenings.
no-castle mode (you can do manually, but auto is better).
pgn-annotation for everymove with N and number of nodes (full+on played move).
autodraw in selfplay when eval=X for Y moves.
ability to create a new weight and train it (maybe too hard or useless).

**Comment by fohristiwhirl** [comment]
@kubiknubika - the only one of those that already exists is "960-chess set seed." - if you know the number, just delete everything in the FEN box and type it in, e.g. 123, and press return. Otherwise, you can type the pieces, e.g. "RKRBBQNN"

**Comment by Debneil** [comment]
Feature request: 
1. auto-analysis of pgn-files with multiple games
2. lichess-style inaccuracy / mistake / blunder counter and centipawn loss for both sides

**Comment by Debneil** [comment]
Feature request: store evaluations and save game as pgn file / append game to another pgn file.

**Comment by fohristiwhirl** [comment]
Well, it's already writing some stats (you can choose which) to a PGN when you save it.

**Comment by Debneil** [comment]
Ah, yes. Really no excuse why I missed that. Just consider the other half of the request, then.

**Comment by koedem** [comment]
I've mentioned a similar idea in the discord some time ago, this would be an even more useful implementation (albeit probably quite a bit of work): For each search store the position and search results in some kind of database. A new search can overwrite the old results based on some criterion (e.g. more nodes searched), whenever the engine is halted and you switch to a new position, show database analysis if existing.

This would be really helpful for deep opening analyses, e.g. when building a repertoire for some opening. You could search a bunch, jump around and later recover all those search results. (including, importantly: the search results for transpositions, even if one didn't search them in this move order)


I realize this would be quite a lot of work, but if it could be added to some long term TODO list that would be great. :)

**Comment by leelanib** [comment]
Te above mentioned graphic is the following: x-axis = moves, y-axis = WDL

**Comment by surh** [comment]
Great app, would it be possible to graph the WDL probability? Something like the graphs in this blog post https://lczero.org/blog/2020/04/wdl-head/

**Comment by bakuldada** [comment]
just a thought based on comments and current wishlist, ability to edit board using drag drop looks clearly the most pressing and must have compared to others.

**Comment by jiho98** [comment]
I agree with the comment above, the board editor clearly looks the most pressing to have. The coordinates looks easiest to implement. And the "show opening names / search from openings by name to immediately set up position." looks nice to have.

**Comment by jiho98** [comment]
I noticed that the last release was on August 22nd. Is Nibbler still in development?

**Comment by fohristiwhirl** [comment]
Not much at the moment.

**Comment by ech1** [comment]
how about an option to load 2 different engines to battle each other ?
idea to see how different elo levels destroy each other

**Comment by mbabigian** [comment]
How about automatically saving the current moves and position when you exit.  I'm used to ChessGUI's I can just close and reopen to the same position / move list as before closing.  Just autosave a pgn on close and reload on startup.

Very grateful for Nibbler, LC0 would be lost without it!

**Comment by mbabigian** [comment]
Also a graceful way to get back to displaying the board/analysis after clicking "Show error Log"

;)

**Comment by toddfreitag** [comment]
I'd like the ability to set per-move node limits.

For example, if I focus 3 different moves and set a limit of 100M nodes per move, after 300M total nodes, each of the 3 focused moves should end up with 100M nodes.

**Comment by yuzisee** [comment]
I know Nibbler isn't really maintained right now, so I don't expect it to get merged any time very soon but for posterity, linking here since it's based on one of the items in the list https://github.com/rooklift/nibbler/pull/238
